<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Philipp Ross" />

<meta name="date" content="2018-02-19" />

<title>Motif search</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 45px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 50px;
  margin-top: -50px;
}

.section h2 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h3 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h4 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h5 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h6 {
  padding-top: 50px;
  margin-top: -50px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">P. berghei mApiAP2</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Motif search</h1>
<h4 class="author"><em>Philipp Ross</em></h4>
<h4 class="date"><em>2018-02-19</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-02-19</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 1e29673</p>
<!-- packages -->
<!-- custom defined functions -->
<div id="overview" class="section level2">
<h2>Overview</h2>
</div>
<div id="preperation" class="section level2">
<h2>Preperation</h2>
<p>What is the genome composition of <em>P. berghei</em>? The following command got me the result seen below:</p>
<pre class="bash"><code>$ seqtk comp PlasmoDB-34_PbergheiANKA_Genome.fasta | awk &#39;BEGIN {l=0;A=0;C=0;G=0;T=0} {l=l+$2;A=A+$3;C=C+$4;G=G+$5;T=T+$6} END {print &quot;A: &quot;A/l&quot; C: &quot;C/l&quot; G: &quot;G/l&quot; T: &quot;T/l}&#39;

A: 0.390528 C: 0.110397 G: 0.110028 T: 0.389048</code></pre>
<p>Now let’s set promoter regions for all the genes we care about. First, we’ll import all genes, filter them, and create lists of genes we want to explore further.</p>
<pre class="r"><code>gff &lt;- rtracklayer::import.gff3(&quot;../data/PlasmoDB-34_PbergheiANKA.gff&quot;)
nuclear_chromosomes &lt;- c(&quot;PbANKA_01_v3&quot;,&quot;PbANKA_02_v3&quot;,&quot;PbANKA_03_v3&quot;,&quot;PbANKA_04_v3&quot;,
                         &quot;PbANKA_05_v3&quot;,&quot;PbANKA_06_v3&quot;,&quot;PbANKA_07_v3&quot;,&quot;PbANKA_08_v3&quot;,
                         &quot;PbANKA_09_v3&quot;,&quot;PbANKA_10_v3&quot;,&quot;PbANKA_11_v3&quot;,&quot;PbANKA_12_v3&quot;,
                         &quot;PbANKA_13_v3&quot;,&quot;PbANKA_14_v3&quot;, &quot;PbANKA_00_v3_archived_contig_1&quot;,
                         &quot;PbANKA_00_v3_archived_contig_2&quot;, &quot;PbANKA_00_v3_archived_contig_3&quot;,
                         &quot;PbANKA_00_v3_archived_contig_4&quot;, &quot;PbANKA_00_v3_archived_contig_5&quot;)
protein_coding_genes &lt;- unique(
  unlist(
    lapply(
      stringr::str_split(
        dplyr::filter(
          tibble::as_tibble(gff),type==&quot;CDS&quot;,seqnames %in% nuclear_chromosomes)$ID,pattern=&quot;[.]&quot;),function(x) x[1])))</code></pre>
<pre><code>Warning in as.data.frame(mcols(x), ...): Arguments in &#39;...&#39; ignored</code></pre>
<pre class="r"><code>genes &lt;- GenomicRanges::GRanges(dplyr::filter(tibble::as_tibble(gff), type==&quot;gene&quot;, ID %in% protein_coding_genes))</code></pre>
<pre><code>Warning in as.data.frame(mcols(x), ...): Arguments in &#39;...&#39; ignored</code></pre>
<p>Now we’ll get promoter intervals:</p>
<pre class="r"><code>proms &lt;- tibble::as_tibble(GenomicRanges::promoters(genes,upstream=1500,downstream=0))</code></pre>
<pre><code>Warning in as.data.frame(mcols(x), ...): Arguments in &#39;...&#39; ignored</code></pre>
<pre class="r"><code>proms$score &lt;- 1
proms$start &lt;- proms$start - 1
proms$start &lt;- ifelse(proms$start&lt;0,0,proms$start)
proms &lt;- dplyr::select(proms,seqnames,start,end,ID,score,strand)
proms &lt;- dplyr::rename(proms,name=ID)
proms$start &lt;- as.integer(proms$start)
readr::write_tsv(x=proms,path=&quot;../data/promoters_1500bps.bed&quot;,col_names=F)</code></pre>
<p>Now use bedtools to generate strand-specific sequences:</p>
<pre class="bash"><code>bedtools getfasta -fi PlasmoDB-34_PbergheiANKA_Genome.fasta -bed promoters_1500bps.bed -s -name &gt; promoters_1500bps.fasta</code></pre>
<p>Unfortunately, I get the following error messages:</p>
<pre class="bash"><code>Feature (PbANKA_01_v3:515305-516805) beyond the length of PbANKA_01_v3 size (515659 bp).  Skipping.
Feature (PbANKA_10_v3:1639716-1641216) beyond the length of PbANKA_10_v3 size (1640193 bp).  Skipping.
Feature (PbANKA_02_v3:621636-623136) beyond the length of PbANKA_02_v3 size (622508 bp).  Skipping.
Feature (PbANKA_06_v3:983178-984678) beyond the length of PbANKA_06_v3 size (984266 bp).  Skipping.
Feature (PbANKA_05_v3:930418-931918) beyond the length of PbANKA_05_v3 size (931174 bp).  Skipping.</code></pre>
<p>I manually went back to correct these features in the original BED file and reran the previous command:</p>
<pre class="bash"><code>bedtools getfasta -fi PlasmoDB-34_PbergheiANKA_Genome.fasta -bed promoters_1500bps_corrected.bed -s -name &gt; promoters_1500bps_corrected.fasta</code></pre>
<p>No warnings this time!</p>
<p>Finally, we can generate background models for these sequences:</p>
<pre class="bash"><code>cat promoters_1500bps_corrected.fasta | fasta-get-markov -m 4 &gt; promoters_1500bps_corrected.background</code></pre>
<p>Finally, we can search the sequences.</p>
</div>
<div id="search-sequences" class="section level2">
<h2>Search sequences</h2>
<p>We’ll use FIMO to search the promoter sequences:</p>
<pre class="bash"><code># search using 8mers
fimo --bgfile data/promoters_1500bps_corrected.background \
  --oc data/fimo_promoters_1500_genomic_1e3_8mer \
  --parse-genomic-coord \
  --thresh 1e-3 \
  data/8mer_motifs.meme \
  data/promoters_1500bps_corrected.fasta 

fimo --bgfile data/promoters_1500bps_corrected.background \
  --oc data/fimo_promoters_1500_1e3_8mer \
  --thresh 1e-3 \
  data/8mer_motifs.meme \
  data/promoters_1500bps_corrected.fasta 

# search using 6mers
fimo --bgfile data/promoters_1500bps_corrected.background \
  --oc data/fimo_promoters_1500_genomic_1e3_6mer \
  --parse-genomic-coord \
  --thresh 1e-3 \
  data/6mer_motifs.meme \
  data/promoters_1500bps_corrected.fasta 

fimo --bgfile data/promoters_1500bps_corrected.background \
  --oc data/fimo_promoters_1500_1e3_6mer \
  --thresh 1e-3 \
  data/6mer_motifs.meme \
  data/promoters_1500bps_corrected.fasta 

# search using 4mers
fimo --bgfile data/promoters_1500bps_corrected.background \
  --oc data/fimo_promoters_1500_genomic_1e2_4mer \
  --parse-genomic-coord \
  --thresh 1e-2 \
  data/4mer_motifs.meme \
  data/promoters_1500bps_corrected.fasta 

fimo --bgfile data/promoters_1500bps_corrected.background \
  --oc data/fimo_promoters_1500_1e2_4mer \
  --thresh 1e-2 \
  data/4mer_motifs.meme \
  data/promoters_1500bps_corrected.fasta </code></pre>
<p>Now we can look at our genes of interest to see if there are any differences between the two motifs:</p>
<pre class="r"><code># import differentially expressed gene list
diff_reg &lt;- readr::read_tsv(&quot;../data/differentially_regulated.txt&quot;,col_names=F)$X1</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character()
)</code></pre>
<pre class="r"><code># remove ribosomal RNA
diff_reg &lt;- diff_reg[!diff_reg %in% &quot;PBANKA_0622941&quot;]
#read in FIMO motif hits
fimo &lt;- readr::read_tsv(&quot;../data/fimo_promoters_1500_1e2_4mer/fimo.txt&quot;,col_names=T)</code></pre>
<pre><code>Parsed with column specification:
cols(
  `# motif_id` = col_character(),
  motif_alt_id = col_character(),
  sequence_name = col_character(),
  start = col_integer(),
  stop = col_integer(),
  strand = col_character(),
  score = col_double(),
  `p-value` = col_double(),
  `q-value` = col_double(),
  matched_sequence = col_character()
)</code></pre>
<pre class="r"><code>fimo &lt;- dplyr::rename(fimo, motif=`# motif_id`)
fimo$sequence_name &lt;- unlist(lapply(stringr::str_split(string=fimo$sequence_name,pattern=&quot;[::]&quot;),function(x) x[1]))
# filter for hits within the genes we care about
fimo_diff_reg &lt;- dplyr::filter(fimo, sequence_name %in% diff_reg)</code></pre>
<p>Now let’s compare the hits within the promoters of our genes of interest to the amount of motifs found in promoters throughout the rest of the genome:</p>
<pre class="r"><code># make a histogram of the data
hist(table(fimo$sequence_name),breaks=20,col=&quot;grey&quot;,main=&quot;Motif Hits Per Promoter&quot;,xlab=&quot;Hits&quot;)
abline(v=mean(table(fimo$sequence_name)),col=&quot;green&quot;,lwd=2)
abline(v=mean(table(fimo_diff_reg$sequence_name)),col=&quot;blue&quot;,lwd=2)
abline(v=max(table(fimo_diff_reg$sequence_name)),col=&quot;red&quot;,lwd=2)</code></pre>
<p><img src="figure/motif_search.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Here, <strong>green</strong> is the average number of hits within the distribution, <strong>blue</strong> is the average number of hits within the differentially regulated genes, and <strong>red</strong> is the gene with the maximum number of motif hits within the differentially expressed gene set.</p>
<p>Let’s save that to a file.</p>
<pre class="r"><code># show data and write it to a file
fimo_diff_reg</code></pre>
<pre><code># A tibble: 894 x 10
   motif motif_alt_id sequence_name  start  stop strand score `p-value`
   &lt;chr&gt; &lt;chr&gt;        &lt;chr&gt;          &lt;int&gt; &lt;int&gt; &lt;chr&gt;  &lt;dbl&gt;     &lt;dbl&gt;
 1 GCTT  &lt;NA&gt;         PBANKA_1246600    32    35 -       8.81   0.00164
 2 GCTT  &lt;NA&gt;         PBANKA_1146600    33    36 -       8.81   0.00164
 3 GCTT  &lt;NA&gt;         PBANKA_1300096    34    37 -       8.81   0.00164
 4 GCTT  &lt;NA&gt;         PBANKA_0722200    37    40 -       8.81   0.00164
 5 GCTT  &lt;NA&gt;         PBANKA_0100021    43    46 -       8.81   0.00164
 6 GCTT  &lt;NA&gt;         PBANKA_0800041    53    56 -       8.81   0.00164
 7 GCTT  &lt;NA&gt;         PBANKA_1365680    74    77 -       8.81   0.00164
 8 GCTT  &lt;NA&gt;         PBANKA_0216000    75    78 -       8.81   0.00164
 9 GCTT  &lt;NA&gt;         PBANKA_0944041   148   151 -       8.81   0.00164
10 GCTT  &lt;NA&gt;         PBANKA_0200600   161   164 -       8.81   0.00164
# ... with 884 more rows, and 2 more variables: `q-value` &lt;dbl&gt;,
#   matched_sequence &lt;chr&gt;</code></pre>
<pre class="r"><code>readr::write_tsv(x=fimo_diff_reg,path=&quot;../output/fimo_differentially_expressed_4mer.tsv&quot;)</code></pre>
</div>
<div id="how-does-this-compare-to-some-kind-of-background" class="section level2">
<h2>How does this compare to some kind of background?</h2>
<pre class="r"><code># sample from our genes to compare to our differentially expressed set of genes
rt &lt;- tibble::tibble(Var1=character(),n=integer())
for (i in 1:1000) {
  s &lt;- sample(x=protein_coding_genes[!protein_coding_genes %in% diff_reg],size=45,replace=F)
  t &lt;- dplyr::filter(fimo, sequence_name %in% s)
  rt &lt;- dplyr::bind_rows(rt,tibble::as_tibble(table(t$motif)))
}

ggplot2::ggplot(rt,ggplot2::aes(x=Var1,y=n)) + 
  geom_boxplot() + 
  geom_hline(yintercept=135,col=&quot;blue&quot;) + 
  geom_hline(yintercept=139,col=&quot;red&quot;)</code></pre>
<p><img src="figure/motif_search.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="what-does-the-concensus-motif-look-like" class="section level2">
<h2>What does the concensus motif look like?</h2>
<p>For the genes that have been identified as differentially expressed, what does the consensus sequence look like? Or better yet, what does the position weight matrix look like?</p>
<pre class="r"><code># num_seqs &lt;- length(fimo_diff_reg$matched)
# 
# # calculate the proportion of nucleotides at each position
# pwm &lt;- tibble::tibble(pos=integer(),A=integer(),C=integer(),G=integer(),T=integer())
# 
# for (i in 1:8) {
#   
#     As &lt;- sum(unlist(lapply(stringr::str_split(fimo_diff_reg$matched,pattern=&quot;&quot;), 
#                  function(x) {x[i] == &quot;A&quot;})))
#     Cs &lt;- sum(unlist(lapply(stringr::str_split(fimo_diff_reg$matched,pattern=&quot;&quot;), 
#                  function(x) {x[i] == &quot;C&quot;})))
#     Ts &lt;- sum(unlist(lapply(stringr::str_split(fimo_diff_reg$matched,pattern=&quot;&quot;), 
#                  function(x) {x[i] == &quot;T&quot;})))
#     Gs &lt;- sum(unlist(lapply(stringr::str_split(fimo_diff_reg$matched,pattern=&quot;&quot;), 
#                  function(x) {x[i] == &quot;G&quot;})))
#   
#   pwm &lt;- dplyr::bind_rows(pwm, tibble::tibble(pos=i,A=As/num_seqs,C=Cs/num_seqs,G=Gs/num_seqs,T=Ts/num_seqs))
# }
# 
# cols1 &lt;- RColorBrewer::brewer.pal(4,&quot;Paired&quot;)
# color_profile &lt;- list(&quot;type&quot; = &quot;per_row&quot;,&quot;col&quot; = c(&quot;#E41A1C&quot;, &quot;#377EB8&quot;, &quot;#F0E442&quot;, &quot;#4DAF4A&quot;))
# 
# logo &lt;- pwm[,2:5]
# p &lt;- slot(seqLogo::makePWM(t(logo)),name=&quot;pwm&quot;)

Logolas::logomaker(fimo_diff_reg$matched_sequence,
                   type=&quot;EDLogo&quot;,
                   bg=c(0.390528,0.110397,0.389048,0.110028)) # calculated  near the start of the document</code></pre>
<pre><code>color_type not provided, so switching to per_row option for color_type</code></pre>
<pre><code>frame width not provided, taken to be 1
frame width not provided, taken to be 1</code></pre>
<p><img src="figure/motif_search.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.1 (2017-06-30)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Gentoo/Linux

Matrix products: default
BLAS: /usr/lib64/blas/reference/libblas.so.3.7.0
LAPACK: /usr/lib64/lapack/reference/liblapack.so.0.0.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bindrcpp_0.2  cowplot_0.9.2 ggplot2_2.2.1

loaded via a namespace (and not attached):
 [1] SummarizedExperiment_1.8.1 lattice_0.20-35           
 [3] Logolas_2.0.2              colorspace_1.3-2          
 [5] htmltools_0.3.6            stats4_3.4.1              
 [7] rtracklayer_1.38.3         yaml_2.1.16               
 [9] utf8_1.1.3                 XML_3.98-1.9              
[11] rlang_0.1.6                pillar_1.1.0              
[13] glue_1.2.0                 BiocParallel_1.12.0       
[15] RColorBrewer_1.1-2         BiocGenerics_0.24.0       
[17] matrixStats_0.53.1         GenomeInfoDbData_1.0.0    
[19] plyr_1.8.4                 bindr_0.1                 
[21] stringr_1.2.0              zlibbioc_1.24.0           
[23] Biostrings_2.46.0          munsell_0.4.3             
[25] gtable_0.2.0               LaplacesDemon_16.1.0      
[27] evaluate_0.10.1            labeling_0.3              
[29] Biobase_2.38.0             knitr_1.19                
[31] IRanges_2.12.0             GenomeInfoDb_1.14.0       
[33] parallel_3.4.1             Rcpp_0.12.15              
[35] readr_1.1.1                scales_0.4.1              
[37] backports_1.0.5            DelayedArray_0.4.1        
[39] S4Vectors_0.16.0           XVector_0.18.0            
[41] Rsamtools_1.30.0           hms_0.4.1                 
[43] digest_0.6.12              stringi_1.1.5             
[45] dplyr_0.7.4                GenomicRanges_1.30.2      
[47] grid_3.4.1                 rprojroot_1.2             
[49] cli_1.0.0                  tools_3.4.1               
[51] bitops_1.0-6               magrittr_1.5              
[53] lazyeval_0.2.1             RCurl_1.95-4.8            
[55] tibble_1.4.2               crayon_1.3.4              
[57] pkgconfig_2.0.1            gridBase_0.4-7            
[59] Matrix_1.2-10              SQUAREM_2017.10-1         
[61] assertthat_0.2.0           rmarkdown_1.8             
[63] R6_2.2.2                   GenomicAlignments_1.14.1  
[65] git2r_0.21.0               compiler_3.4.1            </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
