---
title: "Home"
output:
  html_document:
    toc: false
---

Motif analysis of a polymorphic ApiAP2 protein.

## Motif search

### [Analysis](motif_search.html)

### [Results](motif_search_results.html)

## Methods

Analysis of position weight matrices (PWMs) generated by *in vitro* protein binding microarrays (PBMs) were used to scan *Plasmodium berghei* promoter regions. First, PWMs were trimmed to their 6 most informative base pairs, or their core motifs. PlasmoDB version 34 [1] of the *Plasmodium berghei* ANKA reference genome and gene annotations were downloaded and used to generated promoters of 1500 nucleotides in length. Promoters were defined as 1500 base pairs upstream of the translation start site (ATG) of protein-coding genes only. These promoter regions were then scanned using FIMO version 4.12.0 [2] and a motif hit significance threshold of 0.001. Data analysis was performed in R version 3.4.1 [3]. Code can be found at https://github.com/pzross/pberghei_ap2mutant.

### References

1. PlasmoDB: a functional genomic database for malaria parasites. Nucleic Acids Res. 2008 Oct 31. Aurrecoechea C, et al. 

2. Charles E. Grant, Timothy L. Bailey, and William Stafford Noble, "FIMO: Scanning for occurrences of a given motif", *Bioinformatics* 27(7):1017â€“1018, 2011.

3. R Development Core Team (2008). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. ISBN 3-900051-07-0, URL http://www.R-project.org.
