---
title: "Motify search results"
author: "Philipp Ross"
date: 2018-02-19
output: html_document
editor_options: 
  chunk_output_type: console
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- packages -->
```{r packages, echo=FALSE, include=FALSE}
```

<!-- custom defined functions -->
```{r custom-functions, echo=FALSE}
```

<!-- Add your analysis here -->

```{r import-data,echo=F,include=F}
# prep data
gff <- rtracklayer::import.gff3("../data/PlasmoDB-34_PbergheiANKA.gff")
nuclear_chromosomes <- c("PbANKA_01_v3","PbANKA_02_v3","PbANKA_03_v3","PbANKA_04_v3",
                         "PbANKA_05_v3","PbANKA_06_v3","PbANKA_07_v3","PbANKA_08_v3",
                         "PbANKA_09_v3","PbANKA_10_v3","PbANKA_11_v3","PbANKA_12_v3",
                         "PbANKA_13_v3","PbANKA_14_v3", "PbANKA_00_v3_archived_contig_1",
                         "PbANKA_00_v3_archived_contig_2", "PbANKA_00_v3_archived_contig_3",
                         "PbANKA_00_v3_archived_contig_4", "PbANKA_00_v3_archived_contig_5")
protein_coding_genes <- unique(
  unlist(
    lapply(
      stringr::str_split(
        dplyr::filter(
          tibble::as_tibble(gff),type=="CDS",seqnames %in% nuclear_chromosomes)$ID,pattern="[.]"),function(x) x[1])))
genes <- GenomicRanges::GRanges(dplyr::filter(tibble::as_tibble(gff), type=="gene", ID %in% protein_coding_genes))

# import differentially expressed gene list
diff_reg <- readr::read_tsv("../data/differentially_regulated.txt",col_names=F)$X1
# remove ribosomal RNA
diff_reg <- diff_reg[!diff_reg %in% "PBANKA_0622941"]
# function to read and write fimo data
read_and_write_fimo <- function(input_file,output_file) {
  fimo <- readr::read_tsv(input_file,col_names=T)
  fimo <- dplyr::rename(fimo, motif=`# motif_id`)
  fimo$sequence_name <- unlist(lapply(stringr::str_split(string=fimo$sequence_name,pattern="[::]"),function(x) x[1]))
  # filter for hits within the genes we care about
  fimo_diff_reg <- dplyr::filter(fimo, sequence_name %in% diff_reg)
  # write these to output
  readr::write_tsv(x=fimo,path=output_file)
  readr::write_tsv(x=fimo_diff_reg,path=paste0(output_file,"_differentially_expressed.tsv"))
  # return tibbles
  return(list(f=fimo,d=fimo_diff_reg))
}
# read adn write files
fimo_4mer <- read_and_write_fimo("../data/fimo_promoters_1500_1e2_4mer/fimo.txt","../output/fimo_4mer")
fimo_6mer <- read_and_write_fimo("../data/fimo_promoters_1500_1e3_6mer/fimo.txt","../output/fimo_6mer")
fimo_8mer <- read_and_write_fimo("../data/fimo_promoters_1500_1e3_8mer/fimo.txt","../output/fimo_8mer")
```

## Figure 1

What do the hits within these genes look like if we compare it to hits within the rest of the genome? Let's look at this for 4mers, 6mers, 8mers. Here, **green** is the average number of hits within the distribution, **blue** is the average number of hits within the differentially regulated genes, and **red** is the gene with the maximum number of motif hits within the differentially expressed gene set.

```{r,echo=F}
plot_fimo_hist <- function(fimo_list) {
  hist(table(fimo_list$f$sequence_name),breaks=20,main="",col="grey",xlab="Hits")
  abline(v=mean(table(fimo_list$f$sequence_name)),col="green",lwd=2)
  abline(v=mean(table(fimo_list$d$sequence_name)),col="blue",lwd=2)
  abline(v=max(table(fimo_list$d$sequence_name)),col="red",lwd=2) 
}

plot_fimo_hist(fimo_4mer)
title("4mer Motif Hits Per Promoter")
plot_fimo_hist(fimo_6mer)
title("6mer Motif Hits Per Promoter")
plot_fimo_hist(fimo_8mer)
title("8mer Motif Hits Per Promoter")
```

## Figure 2

If we randomly samply promoters are these motifs over or under represented within our promoters of interest? Let's look at this for 4mers, 6mers, 8mers.

```{r,echo=F}
# sample from our genes to compare to our differentially expressed set of genes
# let's write a function for that
fimo_boxplot <- function(fimo_list) {
  rt <- tibble::tibble(Var1=character(),n=integer())
  for (i in 1:1000) {
    s <- sample(x=protein_coding_genes[!protein_coding_genes %in% diff_reg],size=45,replace=F)
    t <- dplyr::filter(fimo_list$f, sequence_name %in% s)
    rt <- dplyr::bind_rows(rt,tibble::as_tibble(table(t$motif)))
  }

  ggplot(rt,aes(x=Var1,y=n)) + 
    geom_boxplot() + 
    scale_x_discrete(labels=c("ANKA","NK65")) +
    geom_point(aes(x=1,y=135),color="blue",size=4) +
    geom_point(aes(x=2,y=139),color="red",size=4) +
    xlab("") + 
    ylab("Hit Frequency") + 
    ggtitle("Motif Hits per 45\nRandomly Sampled Promoters")
}

fimo_boxplot(fimo_4mer) + xlab("4mer")
fimo_boxplot(fimo_6mer) + xlab("6mer")
fimo_boxplot(fimo_8mer) + xlab("8mer")
```

Here, **red** represents the number of times **GGGCTTAA** is found within our differentially expressed gene set, while **blue** represents the number of times **GGGACTTA** is found within our differentially expressed gene set.

## Figure 3

If we look at the sequences that were matched in the promoters of the identified differentially expressed genes, what does the sequence logo look like if we stacked those sequences on top of one another? 

```{r,echo=F,message=F,warning=F}
fimo_logo <- function(fimo_list) {
  g <- Logolas::logomaker(fimo_list$d$matched_sequence,
                   type="EDLogo",
                   bg=c(0.390528,0.110397,0.389048,0.110028))
  return(g)
}

fimo_logo(fimo_4mer)
fimo_logo(fimo_6mer)
fimo_logo(fimo_8mer)
```

What if we separate them into motif searched for?

### 4mer NK65

```{r,echo=F,message=F,warning=F}
Logolas::logomaker(dplyr::filter(fimo_4mer$d, motif == "GCTT")$matched_sequence,type="EDLogo",bg=c(0.390528,0.110397,0.389048,0.110028))
```

### 4mer ANKA

```{r,echo=F,message=F,warning=F}
Logolas::logomaker(dplyr::filter(fimo_4mer$d, motif == "GACT")$matched_sequence,type="EDLogo",bg=c(0.390528,0.110397,0.389048,0.110028))
```

### 6mer NK65

```{r,echo=F,message=F,warning=F}
Logolas::logomaker(dplyr::filter(fimo_6mer$d, motif == "GGCTTA")$matched_sequence,type="EDLogo",bg=c(0.390528,0.110397,0.389048,0.110028))
```

### 6mer ANKA

```{r,echo=F,message=F,warning=F}
Logolas::logomaker(dplyr::filter(fimo_6mer$d, motif == "GGACTT")$matched_sequence,type="EDLogo",bg=c(0.390528,0.110397,0.389048,0.110028))
```

### 8mer NK65

```{r,echo=F,message=F,warning=F}
Logolas::logomaker(dplyr::filter(fimo_8mer$d, motif == "GGGCTTAA")$matched_sequence,type="EDLogo",bg=c(0.390528,0.110397,0.389048,0.110028))
```

### 8mer ANKA

```{r,echo=F,message=F,warning=F}
Logolas::logomaker(dplyr::filter(fimo_8mer$d, motif == "GGGACTTA")$matched_sequence,type="EDLogo",bg=c(0.390528,0.110397,0.389048,0.110028))
```

## Figure 4

Where are these motifs positionally found within these promoters? First we can look genome wide. The x-axis represents distance from translation start site:

```{r,echo=F,message=F,warning=F}
ggplot(fimo_4mer$f,aes(x=start,group=motif,colour=motif)) + geom_line(stat="density",lwd=2) + ggtitle("4mer\nAll Promoters")
ggplot(fimo_6mer$f,aes(x=start,group=motif,colour=motif)) + geom_line(stat="density",lwd=2) + ggtitle("6mer\nAll Promoters")
ggplot(fimo_8mer$f,aes(x=start,group=motif,colour=motif)) + geom_line(stat="density",lwd=2) + ggtitle("8mer\nAll Promoters")
```

Next we can limit this to just the promoters of the differentially expressed genes:

```{r,echo=F,message=F,warning=F}
ggplot(fimo_4mer$d,aes(x=start,group=motif,colour=motif)) + geom_line(stat="density",lwd=2) + ggtitle("4mer\nDifferentially Expressed Promoters")
ggplot(fimo_6mer$d,aes(x=start,group=motif,colour=motif)) + geom_line(stat="density",lwd=2) + ggtitle("6mer\nDifferentially Expressed Promoters")
ggplot(fimo_8mer$d,aes(x=start,group=motif,colour=motif)) + geom_line(stat="density",lwd=2) + ggtitle("8mer\nDifferentially Expressed Promoters")
```

## Data

### 4mer
```{r,echo=F}
DT::datatable(
  fimo_4mer$d, extensions = 'Buttons', options = list(
    dom = 'Bfrtip',
    buttons = c('copy', 'csv')
  )
)
```

### 6mer
```{r,echo=F}
DT::datatable(
  fimo_6mer$d, extensions = 'Buttons', options = list(
    dom = 'Bfrtip',
    buttons = c('copy', 'csv')
  )
)
```

### 8mer
```{r,echo=F}
DT::datatable(
  fimo_8mer$d, extensions = 'Buttons', options = list(
    dom = 'Bfrtip',
    buttons = c('copy', 'csv')
  )
)
```

## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
